supervisor/supervisor.go

```diff
@@ -0,0 +1,124 @@
+package supervisor
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"github.com/oklog/ulid/v2"
+	"github.com/open-telemetry/opamp-go/client"
+	"github.com/open-telemetry/opamp-go/supervisor/agent"
+	"log"
+	"math/rand"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+type Logger struct {
+	Logger *log.Logger
+}
+
+func (l *Logger) Debugf(format string, v ...interface{}) {
+	l.Logger.Printf(format, v...)
+}
+
+func (l *Logger) Errorf(format string, v ...interface{}) {
+	l.Logger.Printf(format, v...)
+}
+
+type supervisor struct {
```

Comment:
This looks good and roughly the direction that I was thinking we should take with the supervisor.

One question that would be useful to answer: is this generic supervisor flexible enough to allow us to build a specialized Otel Collector Supervisor that knows for example that `--config` flag must be used to specify the location of the config file for the Collector processor and no longer requires the user to specify the `args` settings in the supervisor config file.